### PR TITLE
API - PUT Agent

### PR DESCRIPTION
**Pull request by diogo-s4ntos**
Created method PUT to Agent.

### CODE REVIEW

File: RealestateApi/Controllers/AgentController.cs [hunk 1]
```diff
@@ -95,5 +95,23 @@ public async Task<ActionResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
                 ? Ok(response.Result) 
                 : Problem(response.ProblemType, string.Join(",", response.AdditionalInformation), (int)HttpCodesEnum.BadRequest);
         }
+
+        /// <summary>
+        /// 
+        /// Https Put Method to update an Agent
+        /// 
+        /// </summary>
+        /// <param name="agentId"> Id to update a Agent </param>
```
Comment (Silva096) [inline RealestateApi/Controllers/AgentController.cs:104]: Id to update a Agent ---> Id to update **an** Agent

File: RealestateApi/Controllers/AgentController.cs [hunk 2]
```diff
@@ -95,5 +95,23 @@ public async Task<ActionResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
...
+        /// <param name="newAgentData"> Agent Data to be updated </param>
+        /// <returns> AgentModel </returns>
+        [HttpPut("{agentId}", Name = "PutAgentById")]
+        public async Task<ActionResult<AgentModel>> PutAgenteByIdAsync(int agentId, AgentRequestDto newAgentData)
```
Comment (Silva096) [inline RealestateApi/Controllers/AgentController.cs:108]: PutAgente -> PutAgent

File: RealestateApi/Controllers/AgentController.cs [hunk 3]
```diff
@@ -95,5 +95,23 @@ public async Task<ActionResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
...
+        {
+            var response = await _agentService.PutAgenteByIdAsync(agentId, newAgentData);
```
Comment (Silva096) [inline RealestateApi/Controllers/AgentController.cs:110]: PutAgente -> PutAgent

File: RealestateApi/Repository/AgentRepository.cs [hunk 4]
```diff
@@ -137,5 +137,41 @@ public async Task<bool> DeleteAgentByIdAsync(int agentId)
 
             return affectedRows > 0;
         }
+
+        /// <summary>
+        /// 
+        /// Updates a Agent by Id to the Database
```
Comment (Silva096) [inline RealestateApi/Repository/AgentRepository.cs:143]: Updates "an" agent

File: RealestateApi/Repository/Interfaces/IAgentRepository.cs [hunk 5]
```diff
@@ -39,5 +39,15 @@ public interface IAgentRepository
         /// <param name="agentId"> Id to delete Agent </param>
         /// <returns> AgentModel </returns>
         Task<bool> DeleteAgentByIdAsync(int agentId);
+
+        /// <summary>
+        /// 
+        /// Updates a Agent
```
Comment (Silva096) [inline RealestateApi/Repository/Interfaces/IAgentRepository.cs:45]: an agent

File: RealestateApi/Repository/Interfaces/IAgentRepository.cs [hunk 6]
```diff
@@ -39,5 +39,15 @@ public interface IAgentRepository
...
+        /// 
+        /// </summary>
+        /// <param name="agentId"> Id to update Agent </param>
+        /// <param name="newAgentData"> Agent Data to be Updated </param>
+        /// <returns> AgentModel </returns>
+        Task<AgentModel?> PutAgenteByIdAsync(int agentId, AgentRequestDto newAgentData);
```
Comment (Silva096) [inline RealestateApi/Repository/Interfaces/IAgentRepository.cs:51]: putAgente -> agent

File: RealestateApi/Service/AgentService.cs [hunk 7]
```diff
@@ -1,3 +1,4 @@
+using Microsoft.AspNetCore.Mvc.Diagnostics;
```
Comment (Silva096) [inline RealestateApi/Service/AgentService.cs:1]: Is this needed?

File: RealestateApi/Service/AgentService.cs [hunk 8]
```diff
@@ -160,5 +161,43 @@ public async Task<ServiceResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
             return response;
         }
 
+        /// <summary>
+        /// 
+        /// Updates a Agent by Id
```
Comment (Silva096) [inline RealestateApi/Service/AgentService.cs:166]: An agent

File: RealestateApi/Controllers/AgentController.cs [hunk 9]
```diff
@@ -95,5 +95,23 @@ public async Task<ActionResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
...
+        /// Https Put Method to update an Agent
+        /// 
+        /// </summary>
+        /// <param name="agentId"> Id to update an Agent </param>
+        /// <param name="newAgentData"> Agent Data to be updated </param>
+        /// <returns> AgentModel </returns>
+        [HttpPut("{agentId}", Name = "PutAgentById")]
+        public async Task<ActionResult<AgentModel>> PutAgentByIdAsync(int agentId, AgentRequestDto newAgentData)
+        {
+            var response = await _agentService.PutAgenteByIdAsync(agentId, newAgentData);
```
Comment (RafaelMeha) [inline RealestateApi/Controllers/AgentController.cs:110]: Still "Agente" and not "Agent"

**Review by pmzpro — APPROVED** [review]

**Review by RafaelMeha — APPROVED** [review]

**Pull request merged**